components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -843,12 +845,21 @@ public IdentityProvider getIdentityProvider(String federatedIdPName, String tena
         try {
             startTenantFlow(tenantDomain);
             String claimDialect = ApplicationMgtSystemConfig.getInstance().getClaimDialect();
-            ClaimMapping[] claimMappings = CarbonContext.getThreadLocalCarbonContext().getUserRealm().getClaimManager()
```

Comment:
2. Fix getClaimManager to filter role claim

Reply:
> 1. Test whether `getClaimManager ` uses claim metadata management service.

It does. Therefore added the skipping logic to the relevant method. However, there are multiple places in the claim manager which load local claim but the role claim filtering is only done for the corresponding APIs which are consumed by the UI. Because some APIs are even used by the kernel, changing them actually breaks underline kernel-level flows like getting the user claim values.


> 1. Fix  default ClaimManager to filter role claim

This is done. Please refer to the commits.